### PR TITLE
Polymorphing/Change Staff Rework

### PR DESCRIPTION
**Pull request by Kaedwuff**
Hi! I've been seeing a lot of heat lately about the change staff, and I agree, it's pretty garbo.

So I reworked it, along with the entire polymorph tree of wizard magic.

-Polymorph spells are longer lasting, and when used on an individuals, they do NOT drop all their items, but rather they go away with their older body and are returned to them when they change back
-Polymorphed individuals will have a lot more health than they used to, making it harder to murder-cheese them.  150 hp, in fact, less than a regular humanmob, but also much more than a regular mob of their type
-Polymorphed victims will lose their old mind when polymorphed, becoming like the creature they have  turned into.  If this is a corgi or lizard, they will have to roleplay like one. It is now, however, possible for them to get a hostile form, such as a spider or slime, which means they may attack people. Including the wizard. They get their mind back when they return to their former shape.  
-Wizards can recast Polymorph on someone to change them back to normal early.
-The Base duration of baleful polymorph for wizards is 60 seconds, and it cam be increased up to 180 seconds by using 4 extra spell slots on it, for a bonus 30 seconds each.
-The wand of change works like baleful polymorph, but only lasts 20 seconds.  You can rapidly cause chaos this way, but it is no way permanent, and even the wizard themselves can undo their own transformations with another staff shot.  It also now has 4 shots max, removing the previous nerf put on it due to how horrible it was.
-Polymorphed characters can rename themselves to not be so generic during their tenure as a critter. Technically, anyone nearby can, but it basically works the same way as pet dogs.

Oh, I also removed the utterly garbage bird spell from the code entirely. It was already removed from the spellbooks, no one uses it and it's really bad.

Feedback Thread: https://forums.aurorastation.org/topic/12609-polymorphing-and-staff-of-change-rework/

### CODE REVIEW

**Comment by Kaedwuff** [comment]
!review

File: code/modules/mob/living/simple_animal/friendly/cat.dm [hunk 1]
```diff
@@ -298,3 +298,28 @@
...
+
+/mob/living/simple_animal/cat/wizard
+	desc = "A mysterious feline. How did it get on the station?"
+
+/mob/living/simple_animal/cat/wizard/verb/change_name()
+	set name = "Name Cat"
+	set category = "IC"
+	set src in view(1)
+
+	var/mob/M = usr
```
Comment (skull132) [inline code/modules/mob/living/simple_animal/friendly/cat.dm]:
```suggestion
```

`usr` is already typed as a mob. The variable `M` serves no purpose.

File: code/modules/mob/living/simple_animal/friendly/cat.dm [hunk 2]
```diff
@@ -298,3 +298,28 @@
...
+	if(!M.mind)	return 0
+
+	if(!name_changed)
+
+		var/input = sanitizeSafe(input("What do you want to name the cat?", ,""), MAX_NAME_LEN)
+
+		if(src && input && !M.stat && in_range(M,src))
```
Comment (skull132) [inline code/modules/mob/living/simple_animal/friendly/cat.dm]:
```suggestion
		if(!QDELETED(src) && input && !usr.stat && in_range(usr, src))
```

File: code/modules/mob/living/simple_animal/friendly/cat.dm [hunk 3]
```diff
@@ -298,3 +298,28 @@
...
+/mob/living/simple_animal/cat/wizard
+	desc = "A mysterious feline. How did it get on the station?"
+
+/mob/living/simple_animal/cat/wizard/verb/change_name()
+	set name = "Name Cat"
+	set category = "IC"
+	set src in view(1)
+
+	var/mob/M = usr
+	if(!M.mind)	return 0
```
Comment (skull132) [inline code/modules/mob/living/simple_animal/friendly/cat.dm]:
```suggestion
	if(!usr.mind)
		return 0
```

File: code/modules/mob/living/simple_animal/friendly/corgi.dm [hunk 4]
```diff
@@ -171,3 +171,25 @@
 	if (!stat && !resting && !buckled && prob(1))
 		visible_emote(pick("dances around","chases her tail"),0)
 		INVOKE_ASYNC(src, .proc/do_dance, list(1,2,4,8,4,2,1,2,4,8,4,2,1,2,4,8,4,2))
+
+/mob/living/simple_animal/corgi/verb/change_name()
+	set name = "Name Dog"
+	set category = "IC"
+	set src in view(1)
+
+	var/mob/M = usr
```
Comment (skull132) [inline code/modules/mob/living/simple_animal/friendly/corgi.dm]: Same shit here.
```suggestion
```

File: code/modules/mob/living/simple_animal/friendly/corgi.dm [hunk 5]
```diff
@@ -171,3 +171,25 @@
...
 		visible_emote(pick("dances around","chases her tail"),0)
 		INVOKE_ASYNC(src, .proc/do_dance, list(1,2,4,8,4,2,1,2,4,8,4,2,1,2,4,8,4,2))
+
+/mob/living/simple_animal/corgi/verb/change_name()
+	set name = "Name Dog"
+	set category = "IC"
+	set src in view(1)
+
+	var/mob/M = usr
+	if(!M.mind)	return 0
```
Comment (skull132) [inline code/modules/mob/living/simple_animal/friendly/corgi.dm]:
```suggestion
	if(!usr.mind)
		return 0
```

File: code/modules/mob/living/simple_animal/friendly/corgi.dm [hunk 6]
```diff
@@ -171,3 +171,25 @@
...
+
+	if(!name_changed)
+
+		var/input = sanitizeSafe(input("What do you want to name the dog?", ,""), MAX_NAME_LEN)
+
+		if(src && input && !M.stat && in_range(M,src))
```
Comment (skull132) [inline code/modules/mob/living/simple_animal/friendly/corgi.dm]:
```suggestion
		if(src && input && !usr.stat && in_range(usr, src))
```

File: code/modules/mob/living/simple_animal/friendly/farm_animals.dm [hunk 7]
```diff
@@ -304,3 +304,26 @@
 	name = "emperor penguin"
 	desc = "Emperor of all he surveys."
 
+/mob/living/simple_animal/penguin/wizard
+
+/mob/living/simple_animal/penguin/wizard/verb/change_name()
```
Comment (skull132) [inline code/modules/mob/living/simple_animal/friendly/farm_animals.dm]: tbh, don't duplicate code like this.

Implement a `/mob/living/simple_animal/proc/wizard_change_name(mob/M)` and use the verbs to call that.

**Comment by Geevies** [comment]
@Kaedwuff Still interested?

**Comment by skull132** [comment]
Changes not addressed since September. Closed until updated.

**Pull request closed without merging**